src/module/item/base.ts

```diff
@@ -243,8 +243,45 @@ class ItemPF2e extends Item<ActorPF2e> {
             const { containerId, quantity } = currentSource.system;
             mergeObject(updates, expandObject({ "system.containerId": containerId, "system.quantity": quantity }));
         } else if (currentSource.type === "spell") {
-            // Preserve spellcasting entry location
-            mergeObject(updates, expandObject({ "system.location.value": currentSource.system.location.value }));
+            if (!keepNPCSettings) {
+                // Preserve spellcasting entry location
+                mergeObject(updates, expandObject({ "system.location.value": currentSource.system.location.value }));
```

Comment:
Don't ever use `expandObject`, and just make it do the right thing instead of requiring an argument to be passed.

Reply:
Not sure if I correctly understood what you meant, sorry :-(
So instead of adding an option to do it one way or the other, I should just change it the way I believe it works better?
Regarding expandObject - well, I just copied what was alread there - so instead of doing this I should replace the expandObject entirely and just merge an object I created before?